src/main/java/core/basesyntax/service/impl/MapTransactionService.java

```
@@ -0,0 +1,18 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.TransactionType;
+import core.basesyntax.service.TransactionHandler;
+import core.basesyntax.service.TransactionService;
+import java.util.Map;
+
+public class MapTransactionService implements TransactionService {
+    private final Map<TransactionType, TransactionHandler> handlers;
+
+    public MapTransactionService(Map<TransactionType, TransactionHandler> handlers) {
+        this.handlers = Map.copyOf(handlers);
```

Constructor does not validate its `handlers` parameter. If `handlers` is null callers get an unclear NPE from Map.copyOf. Add an explicit null check and fail fast with an informative message (e.g. `Objects.requireNonNull(handlers, "Handlers map must not be null")`). This addresses the checklist requirement to handle invalid inputs such as "Null parameters".